### PR TITLE
Add a new env_run XML value for E3SM: CASE_GROUP

### LINKED ISSUE #3825: Add a "case-family" argument to create_newcase
In order to complete all the simulations required for DECK, many cases will have to be run.   Each run (pi-control, 20th centruy, SSP85) will start with separate invocations of "create_newcase" with a different argument to "--case" for each simulation and that argument is recorded in the CASE variable in env_run.xml.

For constructing databases and other provenance tools, it would be useful to indicate that each case is part of something larger (in this instance, a DECK run).   The proposal is to add an optional argument like --case-family to create_newcase.  That value would be recorded in env_run.xml under CASE_FAMILY.

So you would have "create_newcase --case picontolCESM3 --case-family DECK_CESM3  --compset ....."

I'm not married to "--case-family" as the name of this argument.  "--simulation-campaign" was also suggested.

### PR DESCRIPTION
**Pull request by jgfouca**
Test suite: by-hand, scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #3825

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review: @rljacob

### CODE REVIEW

**Comment by jgfouca** [comment]
@rljacob , I assume we do not want to tie this value into create_test, just create_newcase?

**Comment by rljacob** [comment]
Yes just create_newcase.

File: scripts/create_newcase [hunk 1]
```diff
@@ -154,6 +154,9 @@ def parse_command_line(args, cimeroot, description):
                         "\nIf provided, the contents of these files will be appended to"
                         "\nthe standard machine files (and any files in ~/.cime).")
 
+    if model == "e3sm":
```
Comment (rljacob) [inline scripts/create_newcase:157]: Don't restrict it to e3sm.  cesm users might find this useful .  As long as its not a required argument.

File: src/drivers/mct/cime_config/config_component_e3sm.xml [hunk 2]
```diff
@@ -795,6 +795,15 @@
     <desc>2**n relative cost of machine (DO NOT EDIT)</desc>
   </entry>
 
+  <entry id="CASE_GROUP">
+    <type>char</type>
+    <group>run_desc</group>
+    <file>env_run.xml</file>
```
Comment (rljacob) [inline src/drivers/mct/cime_config/config_component_e3sm.xml:801]: I think it should be in env_case.xml.   You can't change it without re-running create_newcase.

**Comment by jgfouca** [comment]
@rljacob , I made the changes you wanted. @jedwards4b , I added you as a review since Rob wanted this feature to be opened up to all models.

**Review by rljacob — APPROVED** [review]

**Review by jedwards4b — APPROVED** [review]

**Pull request merged**